tests/unit/src/test/scala/tests/RenameSuite.scala

```diff
@@ -0,0 +1,432 @@
+       |  val str = A.<<renameIt>>("")
+       |}
+       |""".stripMargin,
+    newName = "iAmRenamed",
+    nonOpened = Set(
+      "a/src/main/scala/a/C.scala",
+      "a/src/main/scala/a/D.scala",
+      "a/src/main/scala/a/E.scala"
+    )
+  )
+
+  renamed(
+    "anon",
+    """|/a/src/main/scala/a/Main.scala
+       |trait Methodable[T] {
+       |  def <<method>>(asf: T): Int
+       |}
+       |
+       |trait Alphabet extends Methodable[String] {
+       |  def <<method>>(adf: String) = 123
+       |}
+       |
+       |object Main {
+       |  val a = new Alphabet {
+       |    override def <<me@@thod>>(adf: String): Int = 321
+       |  }
+       |}
+       |""".stripMargin,
+    newName = "renamed"
+  )
+
+  // currently not working due to issues in SemanticDB
+  // renamed(
+  //   "macro-annotation",
+  //   """|/a/src/main/scala/a/Main.scala
+  //      |package a
+  //      |import io.circe.generic.JsonCodec
+  //      |trait LivingBeing
+  //      |@JsonCodec sealed trait <<An@@imal>> extends LivingBeing
+  //      |object Animal {
+  //      |  case object Dog extends <<Animal>>
+  //      |  case object Cat extends <<Animal>>
+  //      |}
+  //      |""".stripMargin,
+  //   newName = "Tree"
+  // )
+  // renamed(
+  //   "classof",
+  //   """|/a/src/main/scala/a/Main.scala
+  //      |package a
+  //      |trait <<A@@BC>>
+  //      |object Main{
+  //      |  val a = classOf[<<A@@BC>>]
+  //      |}
+  //      |""".stripMargin,
+  //   newName = "Animal"
+  // )
+
+  def renamed(
+      name: String,
+      input: String,
+      newName: String,
+      nonOpened: Set[String] = Set.empty
+  ): Unit =
+    check(name, input, newName, notRenamed = false, nonOpened = nonOpened)
+
+  def same(
+      name: String,
+      input: String
+  ): Unit =
+    check(
+      name,
+      input,
+      "SHOULD_NOT_BE_RENAMED",
+      notRenamed = true
+    )
+
+  def check(
+      name: String,
+      input: String,
+      newName: String,
+      notRenamed: Boolean = false,
+      nonOpened: Set[String] = Set.empty
+  ): Unit = {
+    val allMarkersRegex = "(<<|>>|@@|##.*##)"
+    val files = FileLayout.mapFromString(input)
+    val expectedFiles = files.map {
+      case (file, code) =>
+        file -> {
+          if (!notRenamed) {
+            code
+              .replaceAll("\\<\\<\\S*\\>\\>", newName)
+              .replaceAll("##", "")
+          } else {
+            code.replaceAll(allMarkersRegex, "")
+          }
+        }
+    }
+    val base = files.map {
+      case (fileName, code) =>
+        fileName -> code.replaceAll(allMarkersRegex, "")
+    }
+
+    val (filename, edit) = files
+      .find(_._2.contains("@@"))
+      .getOrElse {
+        throw new IllegalArgumentException(
+          "No `@@` was defined that specifies cursor position"
+        )
+      }
+
+    testAsync(name) {
+      cleanWorkspace()
+      for {
+        _ <- server.initialize(
+          s"""/metals.json
+             |{"a":
```

Comment:
Can we add test cases that stress the following scenarios?

* Rename symbol across different build targets
* Rename symbol across Scala/Java (just document that Java files are left unchanged)
* Rename symbol in project with compile error still works (we use an old semanticdb snapshot)

Reply:
Sure, will add that to the tests. It would be super useful to have that explicit.

Reply:
> Rename symbol across different build targets

Added 

> Rename symbol across Scala/Java

Also added

> Rename symbol in project with compile error still works

Added with a type error.